Automattic/wp-calypso
# Replace custom GdprBanner with a8c-analytics.js implementation

#### Proposed Changes

* Remove existing custom Gdpr banner functionality from the code
* Replace the Gdpr banner with a8c-analytics.js unified implementation

#### Testing Instructions (for WIP)

##### Testing plan

* Launch Jetpack Cloud (e.g. `http://jetpack.cloud.localhost:3001/pricing?flags=ad-tracking,gdpr-banner`)
* Remove any existing a8c-analytics cookie (`document.cookie='sensitive_pixel_options=; expires=Thu, 01 Jan 1970 00:00:01 GMT;'; document.cookie='sensitive_pixel_option=; expires=Thu, 01 Jan 1970 00:00:01 GMT;'`)
* Refresh the page
* Notice the cookie banner has been displayed on the page
* Notice that, e.g., GA hasn't been loaded (no `https://www.googletagmanager.com/gtag/js?id=` in page DOM)
* Accept the cookie banner (to simplify the test case, all buckets)
* Refresh the page (or do any action on the page)
* Notice that GA has been loaded now (`https://www.googletagmanager.com/gtag/js?id=` in page DOM)
* When you refresh the page, the cookie banner should no longer be displayed

##### Bonus - test experimental refactored version of a8c-analytics.js (note: Automatticians only)

* In your sandbox, make sure to check out D85697-code diff 
* In your hosts file, point `s1.wp.com` to your sandbox
* Repeat "Testing plan"

#### Pre-merge Checklist






#### To-Do

- [ ] Ensure we can sideload the a8c-analytics.js to wp-calypso without unexpected complications
- [ ] Refactor `setup` file to re-run setup code - as cookie preferences may now change during the runtime

Related to #66493
